tests/test_regressor/test_influence.py

```diff
@@ -0,0 +1,169 @@
+# tests.test_regressor.test_influence
+# Test the regressor influence visualizers.
+#
+# Author:   Benjamin Bengfort <benjamin@bengfort.com>
+# Created:  Sun Jun 09 16:03:31 2019 -0400
+#
+# Copyright (C) 2019 The scikit-yb developers
+# For license information, see LICENSE.txt
+#
+# ID: test_influence.py [] benjamin@bengfort.com $
+
+"""
+Test the regressor influence visualizers.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import pytest
+import numpy as np
+import matplotlib.pyplot as plt
+
+from tests.base import VisualTestCase
+from tests.fixtures import TestDataset
+from sklearn.datasets import make_regression
+
+from yellowbrick.regressor.influence import *
+from yellowbrick.datasets import load_concrete
+
+try:
+    import pandas as pd
+except ImportError:
+    pd = None
+
+
+##########################################################################
+## Fixtures
+##########################################################################
+
+@pytest.fixture(scope='class')
+def data(request):
+    """
+    Creates a random regression fixture that has a R2 score below 0.85 and several
+    outliers that best demonstrate the effectiveness of influence visualizers.
+    """
+    X, y = make_regression(
+        n_samples=100,
+        n_features=14,
+        n_informative=6,
+        bias=1.2,
+        noise=49.8,
+        tail_strength=0.6,
+        random_state=637,
+    )
+
+    request.cls.data = TestDataset(X, y)
+
+
+##########################################################################
+## Assertion Helpers
+##########################################################################
+
+LEARNED_FIELDS = (
+    'distance_', 'p_values_', 'influence_threshold_', 'outlier_percentage_'
+)
+
+
+def assert_not_fitted(oz):
+    for field in LEARNED_FIELDS:
+        assert not hasattr(oz, field)
+
+
+def assert_fitted(oz):
+    for field in LEARNED_FIELDS:
+        assert hasattr(oz, field)
+
+
+##########################################################################
+## Test CooksDistance Visualizer
+##########################################################################
+
+
+@pytest.mark.usefixtures("data")
+class TestCooksDistance(VisualTestCase):
+    """
+    CooksDistance visual test cases
+    """
+
+    def test_cooks_distance(self):
+        """
+        Test image similarity of Cook's Distance on a random dataset
+        """
+        _, ax = plt.subplots()
+        viz = CooksDistance(ax=ax)
+
+        assert_not_fitted(viz)
+        assert viz.fit(self.data.X, self.data.y) is viz
+        assert_fitted(viz)
+
+        # Test fitted values
+        assert viz.distance_.shape == (self.data.X.shape[0],)
+        assert viz.p_values_.shape == viz.distance_.shape
+        assert 0.0 <= viz.influence_threshold_ <= 4.0
+        assert 0.0 <= viz.outlier_percentage_ <= 100.0
```

Comment:
In this case, I'm not testing values but am testing general properties of what I expect the learned attributes to be; this will help us change the random dataset without causing test regressions.